frontend/src/contexts/AuthContext.js

```diff
@@ -17,31 +19,87 @@ export const useAuth = () => {
 
 export const AuthProvider = ({ children }) => {
     const [currentUser, setCurrentUser] = useState()
+    const [currentUserGroup, setCurrentUserGroup] = useState()
     const [loading, setLoading] = useState(true)
     const auth = getAuth(app)
-
+    const whitelist = []
     const provider = new GoogleAuthProvider()
     provider.setCustomParameters({ prompt: 'select_account' })
+    const toast = useToast()
+
+    const showToast = () =>
+        toast({
+            title: 'Invalid email address',
+            description:
+                'Please use a WATonomous or whitelisted email address.',
+            status: 'error',
+            duration: 4000,
+            isClosable: true,
+        })
 
     const login = () => {
         return signInWithPopup(auth, provider)
+            .then((result) => {
+                const { email } = result.user
+                if (whitelist.includes(email)) return
+                if (!email.endsWith('@watonomous.ca')) {
+                    signOut(auth)
+                    showToast()
+                    throw new Error('Invalid email domain')
+                }
+            })
+            .catch((error) => {
+                console.error(error)
+            })
     }
 
     const logout = () => {
         return signOut(auth)
     }
 
+    const setUserGroup = async (user) => {
+        if (user) {
+            const userEmail = user.email
+            const userWatiam = userEmail.substring(0, userEmail.indexOf('@'))
+            const endpoint = `${process.env.REACT_APP_BACKEND_URL}/group/`
+
+            let searchWithEmail = false
+            if (whitelist.includes(userEmail)) searchWithEmail = true //or else use watiam
+
+            try {
+                const retrievedGroup = await axios.post(endpoint, {
+                    field: searchWithEmail ? userEmail : userWatiam,
+                    useEmail: searchWithEmail,
+                })
+                setCurrentUserGroup(retrievedGroup.data.title)
+            } catch (err) {
+                console.log('Error: ' + err)
+            }
+        }
+    }
+
     useEffect(() => {
-        const unsubscribe = onAuthStateChanged(auth, (user) => {
+        const unsubscribe = onAuthStateChanged(auth, async (user) => {
+            if (user) {
+                const email = user.email
+                if (
+                    !email.endsWith('@watonomous.ca') &&
+                    !whitelist.includes(email)
+                )
+                    return
```

Comment:
this should be signOut() as if you just call return you'll get a white screen, I fixed it on master already, I'll do the rebase so you don't need to fix this but just a note